### ISSUE TITLE
"cut: no record found" warnings via "zed lake query" but not "zq"

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2759]
Repro is with Zed commit 0440b15.

While #2749 verified fine, I noticed something else strange in passing that probably deserves a closer look.

Working with the test data [https://github.com/brimdata/zed-sample-data/blob/main/zng/conn.zng.gz,](https://github.com/brimdata/zed-sample-data/blob/main/zng/conn.zng.gz,) I can cut `ts` values from all the records via `zq` without complaint.

```
$ zq -version
Version: v0.29.0-356-g0440b15a

$ zq -z 'cut ts' ~/work/zed-sample-data/zng/conn.zng.gz
{ts:2018-03-24T17:15:21.255387Z}
{ts:2018-03-24T17:15:21.411148Z}
...

$ zq -z 'cut ts' ~/work/zed-sample-data/zng/conn.zng.gz > /dev/null
[no output, so... no errors or warnings]
```

However, if i load the same data into a Pool and run the equivalent Zed query, now I get a bunch of warning messages.

```
$ rm -rf $ZED_LAKE_ROOT

$ zed lake init
using environment variable ZED_LAKE_ROOT
lake created: file:///Users/phil/logs

$ zed lake create -S 5MB -p logs -orderby ts:desc
pool created: logs

$ zed lake load -p logs ~/work/zed-sample-data/zng/conn.zng.gz
1srBN6Qff1zoxMsgffsrKrHNlDy committed 17 segments

$ zed lake query -z 'from logs | cut ts' > /dev/null
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
cut: no record found with columns ts
```

The warnings unfortunately don't give me much to go on in terms of knowing what these records might be that allegedly lack a `ts` field. If I extract the data out in its entirety, once again `zq` doesn't report any problem.

```
$ zed lake query -z 'from logs' > conn.zson

$ ls -l conn.zson
-rw-r--r--  1 phil  staff  375421623 May 21 11:15 conn.zson

$ zq -z 'cut ts' conn.zson > /dev/null
[no output, so... no errors or warnings]
```

**Comment by mccanne** [comment]
I saw this too and believe it happens because the parallelizer creates multiple trunks of the query each with their own cut proc then for inputs that don't have enough scannable segments to give to each parallel paths, the ones that don't get any data complain.

**Comment by mccanne** [comment]
I think we should close this issue and create a new issue to solve this at compile time as the first step toward a generics subsystem for compile-time type checking and field introspection (e.g., for field auto complete in the app).  This new issue can be an epic that we can break into related tasks that build on the basic concept of computing this alongside the seek index at segment write time:
```
count() by typeof(this)
```

**Comment by mccanne** [comment]
Actually instead of closing this outright, we should disable the message for lake queries that have parallelism and put up a PR for this that closes this.

**Comment by philrz** [comment]
Verified in Zed commit 90ac977.

Repeating the repro steps, now instead of the superfluous warning messages, we get the same silence we saw with `zq`, which is appropriate given the particular query.

```
$ zed -version
Version: v0.29.0-361-g90ac9777

$ rm -rf $ZED_LAKE_ROOT

$ zed lake init
using environment variable ZED_LAKE_ROOT
lake created: file:///Users/phil/logs

$ zed lake create -S 5MB -p logs -orderby ts:desc
pool created: logs

$ zed lake load -p logs ~/work/zed-sample-data/zng/conn.zng.gz
1t06a3zUwtI8k8PIwi7anVHNkJK committed 17 segments

$ zed lake query -z 'from logs | cut ts' > /dev/null
$ echo $?
0
```

The longer-term fix that would bring this warning back for `zed query` when appropriate is tracked in #2769. The metadata described there would provide the information necessary to determine up front if the segments of data in the queried Pool entirely lack the fields named in the query and hence provide early feedback to the user before the query is even run, such as in this case informing them that the `cut` would never return anything.

Thanks @mccanne!
